AutoDarkModeSvc/Handlers/HotkeyHandler.cs

```diff
@@ -1,216 +1,204 @@
+            {
+                Register(
+                    builder.Config.Hotkeys.TogglePostpone,
+                    () =>
                     {
-                        state.PostponeManager.AddSkipNextSwitch();
-                        ToastHandler.InvokePauseAutoSwitchToast();
+                        if (builder.Config.AutoThemeSwitchingEnabled)
+                        {
+                            if (state.PostponeManager.IsSkipNextSwitch)
+                            {
+                                state.PostponeManager.RemoveSkipNextSwitch();
+                                ToastHandler.InvokePauseAutoSwitchToast();
+                                Task.Delay(TimeSpan.FromSeconds(2)).ContinueWith(o => ThemeManager.RequestSwitch(new(SwitchSource.Manual)));
+                            }
+                            else
+                            {
+                                state.PostponeManager.AddSkipNextSwitch();
+                                ToastHandler.InvokePauseAutoSwitchToast();
+                            }
+                        }
                     }
-                }
-            });
+                );
+            }
         }
         catch (Exception ex)
         {
-            Logger.Error(ex, "could not register hotkeys:");
+            _logger.Error(ex, "could not register hotkeys:");
         }
-
     }
 
     public static void UnregisterAllHotkeys()
     {
-        Logger.Debug("removing hotkeys");
+        _logger.Debug("removing hotkeys");
         Registered.ForEach(hk => Unregister(hk.Id));
         Registered.Clear();
     }
 
-    public static HotkeyInternal GetRegistered(List<Keys> modifiersPressed, Keys key)
+    public static HotkeyInternal GetRegistered(uint modifiers, uint keyCode)
     {
-        return Registered.Find(hk => Enumerable.SequenceEqual(hk.Modifiers.OrderBy(e => e), modifiersPressed.OrderBy(e => e)) && hk.Key == key);
+        return Registered.Find(hk => hk.Modifiers == modifiers && hk.KeyCode == keyCode);
     }
 
     private static void Register(string hotkeyString, Action action)
     {
         if (Service == null)
         {
-            Logger.Error("service not instantiated while trying to register hotkey");
+            _logger.Error("service not instantiated while trying to register hotkey");
+            return;
         }
-        List<Keys> keys = GetKeyList(hotkeyString);
-        HotkeyInternal mappedHotkey = new() { StringCode = hotkeyString, Action = action };
 
-        keys.ForEach((k) =>
+        try
         {
-            if (IsModifier(k)) mappedHotkey.Modifiers.Add(k);
-            else mappedHotkey.Key = k;
-        });
+            var parts = hotkeyString.Split(',');
+            if (parts.Length != 2 || !uint.TryParse(parts[0], out uint modifiers) || !uint.TryParse(parts[1], out uint keyCode))
+            {
+                _logger.Error($"Invalid hotkey format: {hotkeyString}");
```

Comment:
The hotkey string will map to the numbered keycodes instead of verbose key names. This makes it difficult to read / parse / troubleshoot when looking at the log file.